testament/testament.nim

```diff
@@ -1220,155 +1457,861 @@ proc parseArgs(execState: var Execution, p: var OptParser): ParseCliResult =
 
   execState.userTestOptions = p.cmdLineRest
 
-proc main() =
-  ## Define CLI Options/Args Parsing loops
-  var
-    execState = Execution(flags: {outputColour, logBackend})
-    p = initOptParser()
+if paramCount() == 0:
+  quit Usage
+
+
+const
+  testResultsDir = "testresults"
+  cacheResultsDir = testResultsDir / "cacheresults"
+  noTargetsSpecified: TestTargets = {}
+  defaultExecFlags = {outputColour}
+  defaultBatchSize = 10
+  defaultCatId: CategoryId = 0
+
+func requestedTargets(execState: Execution): set[TTarget] =
+  ## get the requested targets by the user or the defaults
+  if execState.targets == noTargetsSpecified:
+    {targetC, targetJS}
+  else:
+    execState.targets
+
+func `<`(a, b: TestFile): bool {.inline.} =
```

Comment:
It's weird with these not having consistent spacing with its neighbours. Rather than fix the spacing the real takeaway is to move it closer to the type definition.